src/test/munit/imap-secure-trigger-test-case.xml

```diff
@@ -0,0 +1,295 @@
+      xsi:schemaLocation="http://www.mulesoft.org/schema/mule/core http://www.mulesoft.org/schema/mule/core/current/mule.xsd
+      http://www.mulesoft.org/schema/mule/munit-tools  http://www.mulesoft.org/schema/mule/munit-tools/current/mule-munit-tools.xsd
+      http://www.mulesoft.org/schema/mule/validation http://www.mulesoft.org/schema/mule/validation/current/mule-validation.xsd
+      http://www.mulesoft.org/schema/mule/munit http://www.mulesoft.org/schema/mule/munit/current/mule-munit.xsd
+      http://www.mulesoft.org/schema/mule/java http://www.mulesoft.org/schema/mule/java/current/mule-java.xsd
+      http://www.mulesoft.org/schema/mule/tls http://www.mulesoft.org/schema/mule/tls/current/mule-tls.xsd
+      http://www.mulesoft.org/schema/mule/email http://www.mulesoft.org/schema/mule/email/current/mule-email.xsd">
+
+    <configuration-properties file="email-test.properties"/>
+
+    <munit:config name="imaps-trigger-test-case.xml" />
+
+    <!-- Setup/Dispose -->
+
+    <munit:dynamic-port propertyName="imaps.port"/>
+
+    <munit:before-suite name="setUpServer-imaps">
+        <java:invoke-static class="${imap.server}" method="startSecure(Integer)">
+            <java:args>
+                #[{
+                    arg0: ${imaps.port}
+                }]
+            </java:args>
+        </java:invoke-static>
+    </munit:before-suite>
+
+    <munit:before-test name="populate-mailbox-imaps">
+        <foreach collection="#[1 to 10]">
+            <java:invoke-static class="${imap.server}" method="sendEmail()"/>
+        </foreach>
+        <munit-tools:store key="messages">
+            <munit-tools:value>#[[]]</munit-tools:value>
+        </munit-tools:store>
+    </munit:before-test>
+
+    <munit:after-test name="cleanUpServer-imaps">
+        <java:invoke-static class="${imap.server}" method="clean()"/>
+    </munit:after-test>
+
+    <munit:after-suite name="shutDownServer-imaps">
+        <java:invoke-static class="${imap.server}" method="stop()"/>
+    </munit:after-suite>
+
+    <!-- Configs -->
+
+    <email:imap-config name="imaps-config">
+        <email:imaps-connection host="127.0.0.1" port="${imaps.port}" password="password" user="juan.desimoni@mulesoft.com">
+            <tls:context enabledProtocols="TLSv1.2,SSLv3">
+                <tls:trust-store path="greenmail.jks" password="changeit"/>
+            </tls:context>
+        </email:imaps-connection>
+    </email:imap-config>
+
+    <!-- Test messages being polled and assert its content -->
+    <munit:test name="check-messages-content-imaps">
+
+        <munit:enable-flow-sources>
+            <munit:enable-flow-source value="trigger-simple-imaps"/>
+        </munit:enable-flow-sources>
+
+        <munit:execution>
+            <until-successful maxRetries="10" millisBetweenRetries="500">
+                <munit-tools:retrieve key="messages"/>
+                <validation:validate-size value="#[payload]" min="10" max="10"/>
```

Comment:
`<munit-tools:assert-that expression="#[Mtf::queueSize()]" is="#[MunitTools::equalTo(10)]"/>`?